qiskit/circuit/qpy_serialization.py

```diff
@@ -936,8 +1024,96 @@ def _write_instruction(file_obj, instruction_tuple, custom_instructions, index_m
         container.close()
 
 
+def _write_pauli_evolution_gate(file_obj, evolution_gate):
+    operator_list = evolution_gate.operator
+    standalone = False
+    if not isinstance(operator_list, list):
+        operator_list = [operator_list]
+        standalone = True
```

Comment:
Looking at the code of `PauliEvolutionGate`, I'm not super wild about how the type of `.operator` seems to be a bit fluid - feels like this could get buggy fast.  I suppose that's a problem in that class, not this, though.

Reply:
Yeah, I was just trying to work with the constraints of the class. The earlier draft of this just used the `SparsePauliOp` directly before I realized the type hint in the constructor also took a list and it stores it either as a standalone `SparsePauliOp` or a list of them in the class. It would be much cleaner if it was either only a list or only a single. But we need to do this if we're going to faithfully reconstruct the class across qpy